web/client/components/details/Details.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { isEqual } from 'lodash';
+
+import Message from '../I18N/Message';
+import { withConfirmOverride } from '../misc/withConfirm';
+
+import DetailsDialog from './DetailsDialog';
+import DetailsPanel from './DetailsPanel';
+import DetailsSettings from './DetailsSettings';
+
+import DefaultViewer from './viewers/DefaultViewer';
+import QuillEditor from './editors/QuillEditor';
+
+const Details = ({
+    content,
+    editedContent,
+    editorProps = {},
+    settings = {},
+    editedSettings,
+    onSettingsChange,
+    ...props
+}) => {
+    const {editing} = props;
+    const detailsText = editing === 'content' ? editedContent : content;
+
+    const detailsViewer = (<DefaultViewer
+        detailsText={detailsText}/>);
+    const detailsEditor = (<QuillEditor
+        detailsText={detailsText}
+        {...editorProps}/>);
+    const detailsSettings = (<DetailsSettings
+        settings={editing === 'settings' ? editedSettings : settings}
+        onChange={onSettingsChange}/>);
+    const Container = settings.showAsModal ? DetailsDialog : DetailsPanel;
```

Comment:
Hi @vlt1 , I think we should move containers to the plugin level instead of component level.
These are the reasons:
- 1 In general components should fit inside the assigned container. In particular components represented as form/text/lists contents (there are some exception but I don't think this is the case)(*)
- 2 components should not know which container will be assigned to them because see reason in point 1
- 3 we are binding to two specific containers the Details plugin with the proposed implementation this could limit the use of this content in other context (**)

(*) Deatils is a form/text that shows some information about the map and le the user edit the content
(**) there are some other components that use this pattern in MapStore but we should try to improve the responsiveness and pluggability when possible

Reply:
@allyoucanmap What do you mean by moving containers to the plugin level? Making new container plugins akin to BurgerMenu, Toolbar, etc.?

Reply:
@vlt1 No, I don't mean plugin container.
I noticed that within this PR were introduced two components: DetailsDialog and DetailsPanel and they are a variant of ResizeModal and DockPanel. In my opinion these implementations are really specific of plugins/Details.jsx and not the component per se. It would be nice that Details component implements only the content and not container while the container + content can be compose directly in the plugin file.